integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/apiDeploy/DefaultVersionApiTestCase.java

```diff
@@ -0,0 +1,108 @@
+    private final String basePath = "/defaultVersion/";
+    private final String v1 = "1.0.0";
+    private final String v2 = "2.0.0";
+    private final String v1Context = basePath + v1;
+    private final String v2Context = basePath + v2;
+
+    @BeforeClass
+    public void deployAPIs() throws Exception {
+        ApictlUtils.login("test");
+        ApictlUtils.createProject("default_version_v1_OpenAPI.yaml", "default_version_v1", null, null, null,
+                "default_version_v1.yaml");
+        ApictlUtils.createProject("default_version_v2_OpenAPI.yaml", "default_version_v2", null, null, null,
+                "default_version_v2.yaml");
+
+        ApictlUtils.deployAPI("default_version_v1", "test");
+        ApictlUtils.deployAPI("default_version_v2", "test");
+
+        Utils.delay(TestConstant.DEPLOYMENT_WAIT_TIME, "Couldn't wait till deployment completion.");
+
+        API api = new API();
+        api.setName("DefaultVersion");
+        api.setContext(v1Context);
+        api.setVersion(v1);
+        api.setProvider("admin");
+
+        ApplicationDTO app = new ApplicationDTO();
+        app.setName("jwtApp");
+        app.setTier("Unlimited");
+        app.setId((int) (Math.random() * 1000));
+
+        testKeyV1 = TokenUtil.getJWT(api, app, "Unlimited", TestConstant.KEY_TYPE_PRODUCTION, 3600, null, true);
+        api.setContext(v2Context);
+        api.setVersion(v2);
+        testKeyV2 = TokenUtil.getJWT(api, app, "Unlimited", TestConstant.KEY_TYPE_PRODUCTION, 3600, null, true);
+    }
+
+    @Test(description = "Test invoking default versioned API without version in the context")
+    public void testInvokingDefaultVersion() throws CCTestException {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Internal-Key", testKeyV2);
+        HttpResponse response = HttpsClientRequest.doGet("https://localhost:9095" + basePath + "store/inventory", headers);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK, "Response code mismatched");
+        Assert.assertTrue(response.getData().contains("233539"), "Response body mismatched");
+    }
+
+    @Test(description = "Test invoking default versioned API with version in the context")
+    public void testInvokingDefaultVersionWithVersion() throws CCTestException {
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Internal-Key", testKeyV2);
+        HttpResponse response = HttpsClientRequest.doGet("https://localhost:9095" + v2Context + "/store/inventory", headers);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_OK, "Response code mismatched");
+        Assert.assertTrue(response.getData().contains("233539"), "Response body mismatched");
+    }
+
+    @Test(description = "Test invoking `non` default versioned API")
+    public void testInvokingNoneDefaultVersion() throws CCTestException {
```

Comment:
shall we test auth token validation on default API request as well? I guess we have to add throttling and analytics tests as well.

Reply:
You mean a bearer token validation or use v1 token on v2?